src/main/java/tutorlink/parserpackage/Parser.java

```diff
@@ -1,59 +1,136 @@
 package tutorlink.parserpackage;
 
-import tutorlink.commandpackage.studentcommand.AddStudentCommand;
-import tutorlink.commandpackage.studentcommand.DeleteStudentCommand;
-import tutorlink.listpackage.StudentList;
-import tutorlink.studentpackage.StudentClass;
-import exception.StudentNotFoundException;
+import tutorlink.commandpackage.AddCourseCommand;
+import tutorlink.commandpackage.AddStudentCommand;
+import tutorlink.commandpackage.Command;
+import tutorlink.commandpackage.DeleteStudentCommand;
+import tutorlink.commandpackage.ExitCommand;
+import tutorlink.commandpackage.FindStudentCommand;
+import tutorlink.commandpackage.InvalidCommand;
+import tutorlink.commandpackage.ListStudentCommand;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Parser {
+    private static final String EMPTY_INPUT_ERROR_MESSAGE = "Error: Input cannot be empty";
+    private static final String UNKNOWN_COMMAND_ERROR_MESSAGE = "Error: Unknown command";
 
-    /**
-     * Finds a student in the provided student list by matriculation number.
-     *
-     * @param matricNumber the matriculation number of the student to find
-     * @param studentList the list of students to search in
-     * @return the Student object with the given matriculation number
-     * @throws StudentNotFoundException if no student with the given matriculation number is found
-     */
-    private StudentClass findStudent(String matricNumber, StudentList studentList) throws StudentNotFoundException {
-        for (StudentClass a : studentList.getStudentArrayList()) {
-            if (a.getMatricNumber().equals(matricNumber)) {
-                return a;
-            }
+    public Command parse(String line) {
+        if(line.trim().isEmpty()){
+            return new InvalidCommand(EMPTY_INPUT_ERROR_MESSAGE);
+        }
```

Comment:
Perhaps we could assert that `line` should not be empty, since collectUserInput should check for empty line already?

Reply:
Good catch.